cli/gitleaks/gitleaks.py

```
@@ -0,0 +1,320 @@
+            )
+            try:
+                shutil.rmtree(self._job_dir)
+            except OSError as e:
+                self.logger.error(
+                    f"OS error while removing reports download directory: {self._job_dir.absolute()}",
+                )
+                raise e
+
+    @property
+    def logger(self) -> Any:
+        """
+        Instantiates the class logger from the outer scope, unless it is already set.
+        """
+        if not self._logger:
+            self._logger = get_logger(__name__)
+        return self._logger
+
+    @property
+    def detect_args(self) -> list[str]:
+        """
+        Generates the subprocess arguments used by the `gitleaks detect` command.
+
+        Returns:
+            list[str]: a list of strings to pass to the `gitleaks detect` command subprocess.
+        """
+        report = self.gitleaks_report_path.as_posix()
+        path = self.job_dir.as_posix()
+        config_path = self.patterns_file_path.as_posix()
+        return [
+            "--no-git",
+            "--redact",
+            f"--report={report}",
+            f"--path={path}",
+            f"--config-path={config_path}",
+        ]
+
+    @property
+    def token_path(self) -> Path:
+        """
+        Get the token path string set by the `firewatch gitleaks` CLI command as a Path object.
+
+        Returns:
+            Path: The expected path to the file containing the detection patterns server bearer token.
+        """
+        return Path(self._token_path)
+
+    @property
+    def patterns_file_path(self) -> Path:
+        """
+        Get the expected patterns file path as a Path object.
+
+        Returns:
+            Path: The expected path to the file containing the Redhat detection patterns.
+                Defaults to "/tmp/patterns.toml" if not set.
+        """
+        if not self._patterns_file_path:
+            self._patterns_file_path = Path("/tmp/patterns.toml")
+        return self._patterns_file_path
```

why not set default values to the class fields (setting it here hides values from users)? 
and if you decide to leave it like this
```
return self._patterns_file_path or Path("/tmp/patterns.toml")
```

(comment applies to all relevant functions)